ast_canopy/cpp/src/class_template_specialization.cpp

```diff
@@ -0,0 +1,40 @@
+#include <ast_canopy/ast_canopy.hpp>
+
+#include <clang/AST/DeclTemplate.h>
+#include <llvm/ADT/APSInt.h>
+#include <llvm/Support/raw_ostream.h>
+
+namespace ast_canopy {
+
+ClassTemplateSpecialization::ClassTemplateSpecialization(
+    const clang::ClassTemplateSpecializationDecl *CTSD)
+    : Record(llvm::cast<clang::CXXRecordDecl>(CTSD),
+             RecordAncestor::ANCESTOR_IS_NOT_TEMPLATE),
+      class_template(CTSD->getSpecializedTemplate()) {
+
+  // Get the actual template arguments
+  auto &tparam_list = CTSD->getTemplateArgs();
+  actual_template_arguments.reserve(tparam_list.size());
+
+  for (auto i = 0; i < tparam_list.size(); i++) {
+    auto targ = tparam_list[i];
+    clang::TemplateArgument::ArgKind kind = targ.getKind();
+    switch (kind) {
+    case clang::TemplateArgument::ArgKind::Type:
+      actual_template_arguments.push_back(targ.getAsType().getAsString());
+      break;
+    case clang::TemplateArgument::ArgKind::Integral: {
+      llvm::APSInt integer = targ.getAsIntegral();
+      llvm::SmallString<32> str; // -uint64_t::max() is 21 digits (with sign).
```

Comment:
Unsigned 128-bit integer is 39 digits long. So you might want to bump this to 64-bytes and should be sufficient.

Reply:
Does CUDA support 128-bit integers?